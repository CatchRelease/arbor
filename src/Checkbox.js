import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';
import { rem } from 'polished';

import Box from './Box';
import Flex from './Flex';
import Text from './Text';
import Icon from './Icon';

const CheckIcon = ({ fill = 'currentColor', ...props }) => (
  <svg width="10" height="7" viewBox="0 0 10 7" {...props}>
    <path
      fill={fill}
      fillRule="evenodd"
      d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z"
    />
  </svg>
);

CheckIcon.propTypes = {
  fill: PropTypes.string.isRequired
};

const MinusIcon = ({ fill = 'currentColor', ...props }) => (
  <svg width="16" height="16" viewBox="0 0 16 16" {...props}>
    <path
      fill={fill}
      fillRule="evenodd"
      d="M11 7H5c-.55 0-1 .45-1 1s.45 1 1 1h6c.55 0 1-.45 1-1s-.45-1-1-1z"
    />
  </svg>
);

MinusIcon.propTypes = {
  fill: PropTypes.string.isRequired
};

const Label = styled(Flex.withComponent('label'))`
  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
`;

const Input = styled(Box.withComponent('input'))`
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  position: absolute;
  white-space: nowrap;
  width: 1px;
  opacity: 0;
  overflow: hidden;
  padding: 0;
`;

const StyledCheckbox = ({ children, ...props }) => (
  <Flex
    bg="white"
    border="1px solid"
    borderColor="grey60"
    borderRadius="small"
    color="white"
    alignItems="center"
    justifyContent="center"
    width={rem('16px')}
    height={rem('16px')}
    {...props}
  >
    {children}
  </Flex>
);

const Checkbox = ({
  id,
  name,
  label,
  disabled,
  isInvalid,
  checked,
  onChange,
  value,
  indeterminate,
  icon,
  ...props
}) => (
  <Label alignItems="center" my="regular" {...props}>
    <Input
      id={id}
      type="checkbox"
      name={name}
      value={value}
      checked={checked || indeterminate}
      onChange={onChange}
      disabled={disabled}
      aria-invalid={isInvalid}
    />
    {!checked && !disabled && <StyledCheckbox />}
    {checked &&
      !disabled && (
        <StyledCheckbox
          borderColor="blue"
          bg={indeterminate ? 'white' : 'blue'}
          color={indeterminate ? 'blue' : 'white'}
        >
          {indeterminate ? <MinusIcon /> : <CheckIcon />}
        </StyledCheckbox>
      )}
    {disabled && (
      <StyledCheckbox
        borderColor="grey20"
        bg={checked ? 'grey20' : 'white'}
        color={checked ? 'grey50' : 'white'}
      >
        {indeterminate ? <MinusIcon /> : <CheckIcon />}
      </StyledCheckbox>
    )}
    {icon && (
      <Icon
        fontSize="24px"
        name={icon}
        color={disabled ? 'grey20' : 'grey50'}
        ml="smaller"
      />
    )}
    {label && (
      <Text.span
        fontSize="size2"
        color={disabled ? 'grey40' : 'grey100'}
        ml={icon ? 'smallest' : 'smaller'}
      >
        {label}
      </Text.span>
    )}
  </Label>
);

Checkbox.propTypes = {
  /**
   * The id attribute of the checkbox.
   */
  id: PropTypes.string.isRequired,

  /**
   * The name attribute of the checkbox.
   */
  name: PropTypes.string,

  /**
   * Label of the checkbox.
   */
  label: PropTypes.node,

  /**
   * The value attribute of the checkbox.
   */
  value: PropTypes.string,

  /**
   * The checked attribute of the checkbox.
   */
  checked: PropTypes.bool,

  /**
   * State in addition to "checked" and "unchecked".
   * When true, the radio displays a "minus" icon.
   */
  indeterminate: PropTypes.bool,

  /**
   * Function called when state changes.
   */
  onChange: PropTypes.func,

  /**
   * When true, the radio is disabled.
   */
  disabled: PropTypes.bool,

  /**
   * When true, the aria-invalid attribute is true.
   * Used for accessibility.
   */
  isInvalid: PropTypes.bool
};

Checkbox.defaultProps = {
  name: '',
  label: '',
  value: null,
  checked: false,
  indeterminate: false,
  onChange: () => {},
  disabled: false,
  isInvalid: false
};

export default Checkbox;
