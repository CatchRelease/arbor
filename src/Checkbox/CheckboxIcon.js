import React from 'react';
import PropTypes from 'prop-types';

import Icon from '../Icon';

const iconFontSize = '16px';

const CheckboxIcon = ({ checked, indeterminate }) => {
  if (indeterminate) {
    return <Icon name="checkbox-indeterminate" fontSize={iconFontSize} />;
  }

  if (checked) {
    return <Icon name="checkbox-checked" fontSize={iconFontSize} />;
  }

  return null;
};

CheckboxIcon.propTypes = {
  checked: PropTypes.bool,
  indeterminate: PropTypes.bool
};

CheckboxIcon.defaultProps = {
  checked: false,
  indeterminate: false
};

export default CheckboxIcon;
