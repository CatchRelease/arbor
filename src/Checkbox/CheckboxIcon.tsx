import { FC } from 'react';

import Icon from '../Icon';

const iconFontSize = '16px';

type Props = {
  checked?: boolean;
  indeterminate?: boolean;
};

const CheckboxIcon: FC<Props> = ({ checked, indeterminate }) => {
  if (indeterminate) {
    return <Icon name="checkbox-indeterminate" fontSize={iconFontSize} />;
  }

  if (checked) {
    return <Icon name="checkbox-checked" fontSize={iconFontSize} />;
  }

  return null;
};

export default CheckboxIcon;
