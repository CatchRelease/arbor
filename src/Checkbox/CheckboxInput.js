import styled from '@emotion/styled';

import Box from '../Box';

const CheckboxInput = styled(Box)`
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
`;

CheckboxInput.defaultProps = {
  as: 'input'
};

export default CheckboxInput;
