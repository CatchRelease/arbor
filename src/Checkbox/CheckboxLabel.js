import styled from '@emotion/styled';

import Flex from '../Flex';

const CheckboxLabel = styled(Flex.withComponent('label'))`
  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
  line-height: ${({ theme }) => theme.lineHeights.small};
`;

export default CheckboxLabel;
