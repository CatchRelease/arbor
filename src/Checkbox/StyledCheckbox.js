import { css } from '@emotion/core';
import styled from '@emotion/styled';
import { rem } from 'polished';

import Flex from '../Flex';

const baseStyles = ({ theme }) => css`
  background-color: ${theme.colors.white};
  border-radius: ${theme.radii.small};
  border: 1px solid ${theme.colors.grey60};
  color: ${theme.colors.white};
  flex-shrink: 0;
  height: ${rem('16px')};
  width: ${rem('16px')};
`;

const checkedStyles = ({ theme, checked, indeterminate }) =>
  checked &&
  css`
    background-color: ${indeterminate ? theme.colors.white : theme.colors.blue};
    border-color: ${theme.colors.blue};
    color: ${indeterminate ? theme.colors.blue : theme.colors.white};
  `;

const disabledStyles = ({ theme, checked, disabled }) =>
  disabled &&
  css`
    background: ${checked ? theme.colors.grey40 : theme.colors.white};
    border-color: ${theme.colors.border.default};
    color: ${checked ? theme.colors.grey60 : theme.colors.white};
  `;

const StyledCheckbox = styled(Flex)`
  ${baseStyles};
  ${checkedStyles};
  ${disabledStyles};
`;

StyledCheckbox.defaultProps = {
  alignItems: 'center',
  justifyContent: 'center'
};

export default StyledCheckbox;
