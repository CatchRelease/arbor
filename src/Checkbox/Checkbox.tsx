import { ChangeEventHandler, ComponentProps, FC, ReactNode } from 'react';

import Icon from '../Icon';
import Text from '../Text';

import CheckboxIcon from './CheckboxIcon';
import CheckboxInput from './CheckboxInput';
import CheckboxLabel from './CheckboxLabel';
import StyledCheckbox from './StyledCheckbox';

type CheckboxLabelProps = ComponentProps<typeof CheckboxLabel>;

export type Props = CheckboxLabelProps & {
  checked?: boolean;
  disabled?: boolean;
  icon?: string;
  id: string;
  indeterminate?: boolean;
  isInvalid?: boolean;
  label: ReactNode;
  name: string;
  onChange?: ChangeEventHandler<HTMLInputElement>;
  value?: string;
};

const Checkbox: FC<Props> = ({
  checked = false,
  disabled = false,
  icon,
  id,
  indeterminate = false,
  isInvalid = false,
  label,
  name,
  onChange = () => {},
  value = '',
  ...props
}) => (
  <CheckboxLabel alignItems="center" my="regular" {...props}>
    <CheckboxInput
      id={id}
      type="checkbox"
      name={name}
      value={value}
      checked={checked || indeterminate}
      onChange={onChange}
      disabled={disabled}
      aria-invalid={isInvalid}
    />

    <StyledCheckbox {...{ disabled, checked, indeterminate }}>
      <CheckboxIcon {...{ checked, indeterminate }} />
    </StyledCheckbox>

    {icon && (
      <Icon
        fontSize="24px"
        name={icon}
        color={disabled ? 'icon.disabled' : 'icon.default'}
        ml="smaller"
      />
    )}

    <Text
      as="span"
      fontSize="size4"
      color={disabled ? 'text.disabled' : 'text.default'}
      ml={icon ? 'smallest' : 'smaller'}
    >
      {label}
    </Text>
  </CheckboxLabel>
);

export default Checkbox;
