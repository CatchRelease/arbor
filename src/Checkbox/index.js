import React from 'react';
import PropTypes from 'prop-types';

import Icon from '../Icon';
import Text from '../Text';

import CheckboxIcon from './CheckboxIcon';
import CheckboxInput from './CheckboxInput';
import CheckboxLabel from './CheckboxLabel';
import StyledCheckbox from './StyledCheckbox';

const Checkbox = ({
  id,
  name,
  label,
  disabled,
  isInvalid,
  checked,
  onChange,
  value,
  indeterminate,
  icon,
  ...props
}) => (
  <CheckboxLabel alignItems="center" my="regular" {...props}>
    <CheckboxInput
      id={id}
      type="checkbox"
      name={name}
      value={value}
      checked={checked || indeterminate}
      onChange={onChange}
      disabled={disabled}
      aria-invalid={isInvalid}
    />

    <StyledCheckbox {...{ disabled, checked, indeterminate }}>
      <CheckboxIcon {...{ checked, indeterminate }} />
    </StyledCheckbox>

    {icon && (
      <Icon
        fontSize="24px"
        name={icon}
        color={disabled ? 'icon.disabled' : 'icon.default'}
        ml="smaller"
      />
    )}

    <Text.span
      fontSize="size4"
      color={disabled ? 'text.disabled' : 'text.default'}
      ml={icon ? 'smallest' : 'smaller'}
    >
      {label}
    </Text.span>
  </CheckboxLabel>
);

Checkbox.propTypes = {
  /**
   * The checked attribute of the checkbox.
   */
  checked: PropTypes.bool,

  /**
   * When true, the checkbox is disabled.
   */
  disabled: PropTypes.bool,

  /**
   * The id attribute of the checkbox.
   */
  id: PropTypes.string.isRequired,

  /**
   * State in addition to "checked" and "unchecked".
   * When true, the checkbox displays a "minus" icon.
   */
  indeterminate: PropTypes.bool,

  /**
   * When true, the aria-invalid attribute is true.
   * Used for accessibility.
   */
  isInvalid: PropTypes.bool,

  /**
   * Label of the checkbox.
   */
  label: PropTypes.node.isRequired,

  /**
   * The name attribute of the checkbox.
   */
  name: PropTypes.string.isRequired,

  /**
   * Function called when state changes.
   */
  onChange: PropTypes.func,

  /**
   * The value attribute of the checkbox.
   */
  value: PropTypes.string
};

Checkbox.defaultProps = {
  checked: false,
  disabled: false,
  indeterminate: false,
  isInvalid: false,
  onChange: () => {},
  value: ''
};

export default Checkbox;
