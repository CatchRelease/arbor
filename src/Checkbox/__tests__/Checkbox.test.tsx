import createWithTheme from '../../../utils/createWithTheme';
import Checkbox, { Props } from '../Checkbox';

describe('<Checkbox />', () => {
  const baseProps = {
    id: 'example',
    name: 'example',
    label: 'My Checkbox',
    value: '1'
  };

  const createWithProps = (additionalProps?: Partial<Props>) => {
    const props = { ...baseProps, ...additionalProps };

    return createWithTheme(<Checkbox {...{ ...props }} />);
  };

  it('properly renders a checkbox', () => {
    const tree = createWithProps();

    expect(tree).toMatchSnapshot();
  });

  it('properly renders a checked checkbox', () => {
    const tree = createWithProps({ checked: true });

    expect(tree).toMatchSnapshot();
  });

  it('properly renders an indeterminate checkbox', () => {
    const tree = createWithProps({ indeterminate: true });

    expect(tree).toMatchSnapshot();
  });

  it('properly renders a checked and indeterminate checkbox', () => {
    const tree = createWithProps({ indeterminate: true, checked: true });

    expect(tree).toMatchSnapshot();
  });

  it('properly renders a disabled checkbox', () => {
    const tree = createWithProps({ disabled: true });

    expect(tree).toMatchSnapshot();
  });

  it('properly renders a disabled indeterminate checkbox', () => {
    const tree = createWithProps({ disabled: true, indeterminate: true });

    expect(tree).toMatchSnapshot();
  });

  it('properly renders a checkbox with an icon', () => {
    const tree = createWithProps({ icon: 'cr-logo' });

    expect(tree).toMatchSnapshot();
  });
});
