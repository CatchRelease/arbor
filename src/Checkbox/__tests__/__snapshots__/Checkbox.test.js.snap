// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> properly renders a checkbox 1`] = `
.emotion-6 {
  box-sizing: border-box;
  margin-top: 1rem;
  margin-bottom: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  line-height: 1.25;
}

.emotion-0 {
  box-sizing: border-box;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #FFFFFF;
  border-radius: 3px;
  border: 1px solid #989A9C;
  color: #FFFFFF;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 1rem;
  width: 1rem;
}

.emotion-4 {
  font-family: Motiva Sans;
  color: #25272D;
  font-size: 0.875rem;
  margin-left: 0.5rem;
}

<label
  className="emotion-6 emotion-7"
>
  <input
    aria-invalid={false}
    checked={false}
    className="emotion-0 emotion-1"
    disabled={false}
    id="example"
    name="example"
    onChange={[Function]}
    type="checkbox"
    value="1"
  />
  <div
    checked={false}
    className="emotion-2 emotion-3"
    disabled={false}
  />
  <span
    className="emotion-4 emotion-5"
    color="text.default"
    fontSize="size4"
  >
    My Checkbox
  </span>
</label>
`;

exports[`<Checkbox /> properly renders a checkbox with an icon 1`] = `
.emotion-8 {
  box-sizing: border-box;
  margin-top: 1rem;
  margin-bottom: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  line-height: 1.25;
}

.emotion-0 {
  box-sizing: border-box;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #FFFFFF;
  border-radius: 3px;
  border: 1px solid #989A9C;
  color: #FFFFFF;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 1rem;
  width: 1rem;
}

.emotion-4 {
  color: #989A9C;
  font-size: 24px;
  margin-left: 0.5rem;
}

.emotion-6 {
  font-family: Motiva Sans;
  color: #25272D;
  font-size: 0.875rem;
  margin-left: 0.25rem;
}

<label
  className="emotion-8 emotion-9"
>
  <input
    aria-invalid={false}
    checked={false}
    className="emotion-0 emotion-1"
    disabled={false}
    id="example"
    name="example"
    onChange={[Function]}
    type="checkbox"
    value="1"
  />
  <div
    checked={false}
    className="emotion-2 emotion-3"
    disabled={false}
  />
  <i
    className="ar ar-cr-logo emotion-4 emotion-5"
    color="icon.default"
    fontSize="24px"
  />
  <span
    className="emotion-6 emotion-7"
    color="text.default"
    fontSize="size4"
  >
    My Checkbox
  </span>
</label>
`;

exports[`<Checkbox /> properly renders a checked and indeterminate checkbox 1`] = `
.emotion-8 {
  box-sizing: border-box;
  margin-top: 1rem;
  margin-bottom: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  line-height: 1.25;
}

.emotion-0 {
  box-sizing: border-box;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-6 {
  font-family: Motiva Sans;
  color: #25272D;
  font-size: 0.875rem;
  margin-left: 0.5rem;
}

.emotion-2 {
  font-size: 16px;
}

.emotion-4 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #FFFFFF;
  border-radius: 3px;
  border: 1px solid #989A9C;
  color: #FFFFFF;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 1rem;
  width: 1rem;
  background-color: #FFFFFF;
  border-color: #4466EE;
  color: #4466EE;
}

<label
  className="emotion-8 emotion-9"
>
  <input
    aria-invalid={false}
    checked={true}
    className="emotion-0 emotion-1"
    disabled={false}
    id="example"
    name="example"
    onChange={[Function]}
    type="checkbox"
    value="1"
  />
  <div
    checked={true}
    className="emotion-4 emotion-5"
    disabled={false}
  >
    <i
      className="ar ar-checkbox-indeterminate emotion-2 emotion-3"
      fontSize="16px"
    />
  </div>
  <span
    className="emotion-6 emotion-7"
    color="text.default"
    fontSize="size4"
  >
    My Checkbox
  </span>
</label>
`;

exports[`<Checkbox /> properly renders a checked checkbox 1`] = `
.emotion-8 {
  box-sizing: border-box;
  margin-top: 1rem;
  margin-bottom: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  line-height: 1.25;
}

.emotion-0 {
  box-sizing: border-box;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-6 {
  font-family: Motiva Sans;
  color: #25272D;
  font-size: 0.875rem;
  margin-left: 0.5rem;
}

.emotion-4 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #FFFFFF;
  border-radius: 3px;
  border: 1px solid #989A9C;
  color: #FFFFFF;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 1rem;
  width: 1rem;
  background-color: #4466EE;
  border-color: #4466EE;
  color: #FFFFFF;
}

.emotion-2 {
  font-size: 16px;
}

<label
  className="emotion-8 emotion-9"
>
  <input
    aria-invalid={false}
    checked={true}
    className="emotion-0 emotion-1"
    disabled={false}
    id="example"
    name="example"
    onChange={[Function]}
    type="checkbox"
    value="1"
  />
  <div
    checked={true}
    className="emotion-4 emotion-5"
    disabled={false}
  >
    <i
      className="ar ar-checkbox-checked emotion-2 emotion-3"
      fontSize="16px"
    />
  </div>
  <span
    className="emotion-6 emotion-7"
    color="text.default"
    fontSize="size4"
  >
    My Checkbox
  </span>
</label>
`;

exports[`<Checkbox /> properly renders a disabled checkbox 1`] = `
.emotion-6 {
  box-sizing: border-box;
  margin-top: 1rem;
  margin-bottom: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  line-height: 1.25;
}

.emotion-0 {
  box-sizing: border-box;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #FFFFFF;
  border-radius: 3px;
  border: 1px solid #989A9C;
  color: #FFFFFF;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 1rem;
  width: 1rem;
  background: #FFFFFF;
  border-color: #E0E1E2;
  color: #FFFFFF;
}

.emotion-4 {
  font-family: Motiva Sans;
  color: rgba(120,121,125,0.7);
  font-size: 0.875rem;
  margin-left: 0.5rem;
}

<label
  className="emotion-6 emotion-7"
>
  <input
    aria-invalid={false}
    checked={false}
    className="emotion-0 emotion-1"
    disabled={true}
    id="example"
    name="example"
    onChange={[Function]}
    type="checkbox"
    value="1"
  />
  <div
    checked={false}
    className="emotion-2 emotion-3"
    disabled={true}
  />
  <span
    className="emotion-4 emotion-5"
    color="text.disabled"
    fontSize="size4"
  >
    My Checkbox
  </span>
</label>
`;

exports[`<Checkbox /> properly renders a disabled indeterminate checkbox 1`] = `
.emotion-8 {
  box-sizing: border-box;
  margin-top: 1rem;
  margin-bottom: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  line-height: 1.25;
}

.emotion-0 {
  box-sizing: border-box;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-2 {
  font-size: 16px;
}

.emotion-4 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #FFFFFF;
  border-radius: 3px;
  border: 1px solid #989A9C;
  color: #FFFFFF;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 1rem;
  width: 1rem;
  background: #FFFFFF;
  border-color: #E0E1E2;
  color: #FFFFFF;
}

.emotion-6 {
  font-family: Motiva Sans;
  color: rgba(120,121,125,0.7);
  font-size: 0.875rem;
  margin-left: 0.5rem;
}

<label
  className="emotion-8 emotion-9"
>
  <input
    aria-invalid={false}
    checked={true}
    className="emotion-0 emotion-1"
    disabled={true}
    id="example"
    name="example"
    onChange={[Function]}
    type="checkbox"
    value="1"
  />
  <div
    checked={false}
    className="emotion-4 emotion-5"
    disabled={true}
  >
    <i
      className="ar ar-checkbox-indeterminate emotion-2 emotion-3"
      fontSize="16px"
    />
  </div>
  <span
    className="emotion-6 emotion-7"
    color="text.disabled"
    fontSize="size4"
  >
    My Checkbox
  </span>
</label>
`;

exports[`<Checkbox /> properly renders an indeterminate checkbox 1`] = `
.emotion-8 {
  box-sizing: border-box;
  margin-top: 1rem;
  margin-bottom: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  line-height: 1.25;
}

.emotion-0 {
  box-sizing: border-box;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-4 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #FFFFFF;
  border-radius: 3px;
  border: 1px solid #989A9C;
  color: #FFFFFF;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 1rem;
  width: 1rem;
}

.emotion-6 {
  font-family: Motiva Sans;
  color: #25272D;
  font-size: 0.875rem;
  margin-left: 0.5rem;
}

.emotion-2 {
  font-size: 16px;
}

<label
  className="emotion-8 emotion-9"
>
  <input
    aria-invalid={false}
    checked={true}
    className="emotion-0 emotion-1"
    disabled={false}
    id="example"
    name="example"
    onChange={[Function]}
    type="checkbox"
    value="1"
  />
  <div
    checked={false}
    className="emotion-4 emotion-5"
    disabled={false}
  >
    <i
      className="ar ar-checkbox-indeterminate emotion-2 emotion-3"
      fontSize="16px"
    />
  </div>
  <span
    className="emotion-6 emotion-7"
    color="text.default"
    fontSize="size4"
  >
    My Checkbox
  </span>
</label>
`;
